src/utils/planner.ts

```diff
@@ -40,33 +44,125 @@ export const planner = {
   /**
    * Get the Planner plan for a specific Roster.
    * @param rosterId Roster ID.
+   * @param metadata OData metadata level. Default is none.
+   * @throws Error when the roster has no plan.
    */
   async getPlanByRosterId(rosterId: string, metadata: 'none' | 'minimal' | 'full' = 'none'): Promise<PlannerPlan> {
     const plans = await odata.getAllItems<PlannerPlan>(`${graphResource}/beta/planner/rosters/${rosterId}/plans`, metadata);
+
+    if (!plans.length) {
```

Comment:
This apparently works, because `!plans.length` evaluates to `true`. as `0` can evaluate to `false`.
Still, I dislike the slight JavaScript-smell... just like checks where `""` can evaluate to `true`/`false` when used in a condition. I'd prefer if we just use an explicit check in the type that the value:

```suggestion
if (plans.length === 0) {
  //...
}
```